### ISSUE TITLE
Add support for Windows Server 2022 to Docker Hub image

### ISSUE THREAD

**Issue by ldti** [SystemRage/py-kms#139]
Per https://github.com/SystemRage/py-kms/issues/138, Please update the Docker image on Hub when possible with the latest py-kms version, including Server 2022 support.

Thank you in advance.

**Comment by simonmicro** [comment]
Okay, I plan to switch here to use GitHub Actions instead of the Docker Hub infrastructure - I found some notes here: https://blog.oddbit.com/post/2020-09-25-building-multi-architecture-im/

@SystemRage I will need some more permissions to use the Github Actions here - I kindly ask you to extend my permissions so I'm at least able to accept PRs myself for this repository (maybe even (co-)maintainer level). Feel free to PM me so we can chat about that :smiley:

**Comment by thespad** [comment]
If it helps anyone I've done some builds of the minimal image based on this change https://github.com/SystemRage/py-kms/issues/138#issuecomment-871736924 and tested that it works with Server 2022 - though I haven't properly tested the ARM builds.

`docker pull ghcr.io/thespad/py-kms:latest`

Sources etc: https://github.com/TheSpad/docker-py-kms

**Comment by simonmicro** [comment]
@TheSpad Thanks for your work, I'll try to use GA for all different architectures - should be easy enough :P
See my other comment until I'll do this on my own.

**Comment by simonmicro** [comment]
@TheSpad Thanks for your files (https://github.com/TheSpad/docker-py-kms) that I've used for inspiration - the fork (https://github.com/Py-KMS-Organization/py-kms) has now full build support for even more architectures, docker hub updates and GitHub Container Repository Integration. I hope all works now - could someone validate that?

**Comment by dudefoxlive** [comment]
> 
> 
> @TheSpad Thanks for your files (https://github.com/TheSpad/docker-py-kms) that I've used for inspiration - the fork (https://github.com/Py-KMS-Organization/py-kms) has now full build support for even more architectures, docker hub updates and GitHub Container Repository Integration. I hope all works now - could someone validate that?

Will the python files also get updated to support server 2022 along with the docker image at some point in your fork? I don't want to use docker if I don't have to since I am running mine in a Proxmox LXC container.

**Comment by simonmicro** [comment]
According to the last comment on #138 it already works?! But also I've noted the wiki was not up-to-date, therefore I added the keys for 2022 there.

**Comment by thespad** [comment]
> 
> 
> According to the last comment on #138 it already works?! But also I've noted the wiki was not up-to-date, therefore I added the keys for 2022 there.

I don't believe the fix was ever committed. When I did my builds I had to inject my own version of the file with the modifications made.

**Comment by simonmicro** [comment]
Oh, to get that straight: All needed for the 2022 Server to work is just that one line inside 164 at `./py-kms/pykms_Base.py`?

**Comment by simonmicro** [comment]
And it does not break anything else? In that case I'll accept any PR or add it myself - but only if anyone could confirm that for me (I do not want to break everything :grin:  ).

**Comment by thespad** [comment]
I can't say 100% it won't break anything else because I only tested with Server 2022 but it worked fine for me https://github.com/TheSpad/docker-py-kms/blob/main/pykms_Base.py#L164

**Comment by simonmicro** [comment]
Okay, fix is added and pushed - let's hope for the best!
